### PR TITLE
Improve documentation related for particle subemitters, collision and attractors

### LINKED ISSUE #83913: GPU Particles 3D has no documentation about using Visibility AABB for collisions
### Godot version

v4.1.2.stable; v4.2.beta3

### System information

Godot v4.2.beta2 - Windows 10.0.19045 - Vulkan (Forward+, Mobile) - GLES3 (Compatibility) - dedicated NVIDIA GeForce GTX 1650 (NVIDIA; 31.0.15.4584) - Intel(R) Core(TM) i7-2600K CPU @ 3.40GHz (8 Threads)

### Issue description

It took me a while to figure out that **Drawing > Visibility AABB** also controls the GPUParticle3D's area of collision. I almost thought it was a bug.

This behavior is different from GPUParticle2D, where the Visibility Rect doesn't affect collisions.

It is also not documented, unlike how LightOccluder2D being used for 2D GPU collisions is well explained.

My request is to add some documentation about this on the 3D particle's collision settings and/or the Visibility AABB property.

Something simple like:
- "Note: 3D Particle's collisions are only handled by GPUParticlesCollision3D nodes touching the Visibility AABB area"
- "Grow the box to define the area your particles should detect GPUParticlesCollision3D nodes for collisions."

GPUParticle3D Collision Hints

### Steps to reproduce

The reproduction project has "main_3d" scene that shows how Visibility AABB controls the collisions, even in the editor.
There is also a "main_2d" to show how 2D behaves, for reference.

### Minimal reproduction project

[GPUParticle3D Collisions AABB.zip](https://github.com/godotengine/godot/files/13160849/GPUParticle3D.Collisions.AABB.zip)

### PR DESCRIPTION
**Pull request by Calinou**
This also describes how to set up particle flipbook animation in GPUParticles2D's texture property.

- This closes https://github.com/godotengine/godot/issues/83913.

### CODE REVIEW

File: doc/classes/GPUParticles2D.xml [hunk 1]
```diff
@@ -101,10 +103,12 @@
 			Particle system's running speed scaling ratio. A value of [code]0[/code] can be used to pause the particles.
 		</member>
 		<member name="sub_emitter" type="NodePath" setter="set_sub_emitter" getter="get_sub_emitter" default="NodePath(&quot;&quot;)">
-			The [NodePath] to the [GPUParticles2D] used for sub-emissions.
+			Path to another [GPUParticles2D] node that will be used as a subemitter (see [member ParticleProcessMaterial.sub_emitter_mode]). Subemitters can be used to achieve effects such as fireworks, sparks on collision, bubbles popping into water drops, and more.
+			[b]Note:[/b] When [member sub_emitter] is set, the target [GPUParticles3D] node will no longer emit particles on its own.
```
Comment (mrcdk) [inline doc/classes/GPUParticles2D.xml:107]: Typo: `[GPUParticles3D]` -> `[GPUParticles2D]`

File: doc/classes/ParticleProcessMaterial.xml [hunk 2]
```diff
@@ -124,11 +124,11 @@
...
 			The particles' collision mode.
-			[b]Note:[/b] 3D Particles can only collide with [GPUParticlesCollision3D] nodes, not [PhysicsBody3D] nodes. To make particles collide with various objects, you can add [GPUParticlesCollision3D] nodes as children of [PhysicsBody3D] nodes.
+			[b]Note:[/b] 3D Particles can only collide with [GPUParticlesCollision3D] nodes, not [PhysicsBody3D] nodes. To make particles collide with various objects, you can add [GPUParticlesCollision3D] nodes as children of [PhysicsBody3D] nodes. In 3D, collisions only occur within the area defined by the [GPUParticles3D] node's [member GPUParticles3D.visibility_aabb].
 			[b]Note:[/b] 2D Particles can only collide with [LightOccluder2D] nodes, not [PhysicsBody2D] nodes.
 		</member>
 		<member name="collision_use_scale" type="bool" setter="set_collision_use_scale" getter="is_collision_using_scale" default="false">
-			Should collision take scale into account.
+			If [code]true[/code], [member GPUParticles3D.collision_base_size] is multiplied by the particle's effective scale (see [member scale_min], [member scale_max], [member scale_curve] and [member scale_over_velocity_curve]).
```
Comment (AThousandShips) [inline doc/classes/ParticleProcessMaterial.xml:131]:
```suggestion
			If [code]true[/code], [member GPUParticles3D.collision_base_size] is multiplied by the particle's effective scale (see [member scale_min], [member scale_max], [member scale_curve], and [member scale_over_velocity_curve]).
```

File: doc/classes/GPUParticles2D.xml [hunk 3]
```diff
@@ -101,10 +103,12 @@
...
 		</member>
 		<member name="texture" type="Texture2D" setter="set_texture" getter="get_texture">
-			Particle texture. If [code]null[/code], particles will be squares.
+			Particle texture. If [code]null[/code], particles will be squares with a size of 1×1 pixels.
+			[b]Note:[/b] To use a flipbook texture, assign a new [CanvasItemMaterial] to the [GPUParticles2D]'s [member CanvasItem.material] property, then enable [member CanvasItemMaterial.particles_animation] and set [member CanvasItemMaterial.particles_anim_h_frames], [member CanvasItemMaterial.particles_anim_v_frames] and [member CanvasItemMaterial.particles_anim_loop] to match the flipbook texture.
```
Comment (AThousandShips) [inline doc/classes/GPUParticles2D.xml:111]:
```suggestion
			[b]Note:[/b] To use a flipbook texture, assign a new [CanvasItemMaterial] to the [GPUParticles2D]'s [member CanvasItem.material] property, then enable [member CanvasItemMaterial.particles_animation] and set [member CanvasItemMaterial.particles_anim_h_frames], [member CanvasItemMaterial.particles_anim_v_frames], and [member CanvasItemMaterial.particles_anim_loop] to match the flipbook texture.
```

**Review by QbieShay — APPROVED** [review]
Looks good! Thank you 

I was a bit concerned by the mention of flipbook because I wasn't sure how standard it is, but it does seem pretty standard.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**